add a check and only set up and build compiler-rt builtins if they have
not been disabled
Index: llvm-toolchain-snapshot_16~++20220920091612+a7d2409bac7f/llvm/runtimes/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_16~++20220920091612+a7d2409bac7f.orig/llvm/runtimes/CMakeLists.txt
+++ llvm-toolchain-snapshot_16~++20220920091612+a7d2409bac7f/llvm/runtimes/CMakeLists.txt
@@ -124,7 +124,12 @@ endfunction()
 # is required because the other runtimes need the builtin libraries present
 # before the just-built compiler can pass the configuration tests.
 get_compiler_rt_path(compiler_rt_path)
-if(compiler_rt_path)
+# check if builtins have been disabled
+set(RUNTIMES_BUILD_BUILTINS ON)
+if(DEFINED COMPILER_RT_BUILD_BUILTINS)
+  set(RUNTIMES_BUILD_BUILTINS ${COMPILER_RT_BUILD_BUILTINS})
+endif()
+if(compiler_rt_path AND RUNTIMES_BUILD_BUILTINS)
   if(NOT LLVM_BUILTIN_TARGETS)
     builtin_default_target(${compiler_rt_path}
       DEPENDS clang-resource-headers)
